scirpy/_plotting/_clonotype_imbalance.py

```diff
@@ -0,0 +1,162 @@
+    additional_hue
+        An additional grouping factor. If the `case_label` was tumor for example, this could
+        help make a distinction between imbalance in lung and colorectal tumors.
+    top_n
+        The number of top clonotypes to be visualized.
+    fraction
+        If `True`, compute fractions of abundances relative to the `groupby` column
+        rather than reporting abosolute numbers. Alternatively, a column 
+        name can be provided according to that the values will be normalized or an iterable
+        providing cell weights directly. Setting it to `False` or `None` assigns equal weight
+        to all cells.
+    plot_type
+        Whether a volcano plot of statistics or a box/bar/strip plot of frequencies should be shown.
+    inplace
+        Whether results should be added to `uns` or returned directly.
+    added_key
+        If the tools has already been run, the results are added to `uns` under this key.
+    **kwargs
+        Additional arguments passed to the base plotting function.  
+    
+    Returns
+    -------
+    Axes object
+    """
+
+    if added_key not in adata.uns:
+        tl.clonotype_imbalance(
+            adata,
+            replicate_col=replicate_col,
+            groupby=groupby,
+            case_label=case_label,
+            control_label=control_label,
+            target_col=target_col,
+            additional_hue=additional_hue,
+            fraction=fraction,
+            added_key=added_key,
+        )
+
+    df = adata.uns[added_key]["pvalues"]
+
+    if plot_type == "volcano":
+        df = df.loc[:, ["logFC", "logpValue"]]
+        default_style_kws = {"title": title, "xlab": xlab, "ylab": ylab}
+        if "style_kws" in kwargs:
+            default_style_kws.update(kwargs["style_kws"])
+        kwargs["style_kws"] = default_style_kws
+        return volcano(df, **kwargs)
+
+    else:
+        df = df.sort_values(by="pValue")
+        df = df.head(n=top_n)
+
+        tclt_df = adata.uns[added_key]["abundance"]
+        tclt_df = tclt_df.loc[tclt_df[target_col].isin(df[target_col]), :]
+
+        if additional_hue is None:
+            tclt_df = tclt_df.pivot_table(
+                index=[groupby, replicate_col],
+                columns=target_col,
+                values="Normalized abundance",
+                fill_value=0,
+            ).reset_index()
+            tclt_df = pd.melt(
+                tclt_df,
+                id_vars=[groupby, replicate_col],
+                value_name="Normalized abundance",
+            )
+            if plot_type == "box":
+                ax = sns.boxplot(
+                    x=target_col, y="Normalized abundance", hue=groupby, data=tclt_df
+                )
+            else:
+                if plot_type == "bar":
+                    ax = sns.barplot(
+                        x=target_col,
+                        y="Normalized abundance",
+                        hue=groupby,
+                        data=tclt_df,
```

Comment:
I am not sure I got it right. Added a fig_kws argument, but that seems like a duplicate of the kast **kwargs...  
On the other hand, I also grouped the four main arguments so that we are not that verbose.